src/clients/UBAClient/UBAClientTypes.ts

```diff
@@ -1,9 +1,9 @@
 import { BigNumber } from "ethers";
-import { UbaFlow } from "../../interfaces";
+import { FillWithBlock, UbaFlow } from "../../interfaces";
 import { UBAActionType, FlowTupleParameters } from "../../UBAFeeCalculator/UBAFeeTypes";
 import UBAConfig from "../../UBAFeeCalculator/UBAFeeConfig";
 
-export type RequestValidReturnType = { valid: false; reason: string } | { valid: true };
+export type RequestValidReturnType = { valid: boolean; reason?: string; matchingFill?: FillWithBlock };
```

Comment:
Agree this is good, and might also be worth returning the corresponding deposit as well?  Given that we already do the work to look it up...

Reply:
I want to be practical. We don't need it as far as I can tell in the dataworker. Does the Relayer needd it from this function?